homeassistant/components/intellifire/config_flow.py

```diff
@@ -0,0 +1,74 @@
+"""Config flow for IntelliFire integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from intellifire4py import IntellifireAsync
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import DOMAIN, LOGGER
+
+STEP_USER_DATA_SCHEMA = vol.Schema({vol.Required(CONF_HOST): str})
+
+
+async def validate_input(hass: HomeAssistant, host: str) -> str:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+    api = IntellifireAsync(host)
+    try:
+        await api.poll()
+    except ConnectionError as exception:
+        raise CannotConnect from exception
```

Comment:
You could remove the try / catch here, since `CannotConnect` is a helper exception you defined yourselves. If you don't do any try/catch in `validate_input`, than your try catch in Config Flow (line 50) will catch it and there you could catch your `ConnectionError` and set `errors["base"] = "cannot_connect"`.